include/data/nodeid.h

```diff
@@ -54,5 +54,9 @@ class NodeID {
   explicit operator bool() const;
 };
 
+struct NodeIDHash {
```

Comment:
This is a matter of taste. Specialization of std::hash requires extending std namespace.